src/dune/foreign_rules.ml

```diff
@@ -72,64 +72,52 @@ let include_dir_flags ~expander ~dir (stubs : Foreign.Stubs.t) =
          in
          Command.Args.S [ A "-I"; Path include_dir; dep_args ]))
 
-let build_c_file ~sctx ~dir ~expander ~include_flags (loc, src, dst) =
-  let flags = Foreign.Source.flags src in
+let build_c_or_cxx_file ~language ~sctx ~dir ~expander ~include_flags
+    (loc, src, dst) =
   let ctx = Super_context.context sctx in
-  let c_flags =
-    Super_context.foreign_flags sctx ~dir ~expander ~flags
-      ~language:Foreign.Language.C
+  let flags =
+    let ctx_flags =
+      match language with
+      | Foreign.Language.C ->
+        let cfg = ctx.ocaml_config in
+        List.concat
+          [ Ocaml_config.ocamlc_cflags cfg
+          ; Ocaml_config.ocamlc_cppflags cfg
+          ; Fdo.c_flags ctx
+          ]
+      | Foreign.Language.Cxx -> Fdo.cxx_flags ctx
```

Comment:
Does that mean that we don't pass the `ocamlc_cflags` to the C++ compiler anymore? I thought we were passing them before, except for the `-std=...` ones.

Reply:
I have restored the current behaviour for C++ flags:
https://github.com/ocaml/dune/pull/3565/files#diff-fa2be1f0411c09542cd1046d808c7b3dR13-R14